### DISCUSSION TITLE
New 'Lr' Scheduler Issue

### DISCUSSION THREAD

**Discussion by sakerlrf** [open-mmlab/mmaction2#1175] [Q&A] [answered]
Hello,

Trying to add [`RelativeStepLrUpdaterHook`](https://mmaction2.readthedocs.io/en/latest/tutorials/5_new_modules.html?highlight=policy#add-new-learning-rate-scheduler-updater), but I am confused with the folders in the _**core**_. 
- In this current rep online, I see that inside the _**core**_, there is no `lr` folder. Instead, there is only the `scheduler`, holding the `lr_updater.py`; having a general name even though it refers to specific scheduler!?
- In the [docs](https://mmaction2.readthedocs.io/en/latest/tutorials/5_new_modules.html#add-new-learning-rate-scheduler-updater), the path is suggested: **_$MMAction2/mmaction/core/lr_** 

While in [mmcv/mmcv/runner/hooks/lr_updater.py](https://github.com/open-mmlab/mmcv/blob/f48241a65aebfe07db122e9db320c31b685dc674/mmcv/runner/hooks/lr_updater.py), the 'lr_updater.py' has all the registered schedulers. 

Adding the scheduler
-------------------------
For the addition, the steps I took were to create the **RelativeStep_lr_hook.py**, in "mmaction/core/lr" and import the enclosed class to `__init__.py` in the same dir.
When I set the **_lr_config_**, in the model config script and try to train a model, I get:

`TypeError: RelativeStepLrUpdaterHook: __init__() missing 1 required positional argument: 'runner'`

Any input would be helpful.
Thanks!

**Comment by dreamerlin** [comment]
1. Thanks for your report. the dir has changed to `core/scheduler`. We will update it soon.
2. you forget to pass runner to `__init__`.

**Reply by sakerlrf** [reply]
Thank you for your answer @dreamerlin. Here is what I have in the **__init__.py**:

`from .RelativeStep_lr_hook import RelativeStepLrUpdaterHook`
`__all__ = ['RelativeStepLrUpdaterHook']`

**Reply by sakerlrf** [reply]
@dreamerlin, Could you please demonstrate what should be added?
Do you possibly refer to the 'base_runner'?

Thank you.

**Reply by dreamerlin** [reply]
Oops ... The `runner` should not be in `__init__`. Thanks for your  report, we will update it

**Comment by dreamerlin** [comment, marked as answer]
https://github.com/open-mmlab/mmaction2/pull/1169/commits/9dab3319fb042678f91970534cf04c409c677571
